source/src/cip/ciptypes.h

```diff
@@ -275,12 +276,18 @@ typedef struct {
 typedef void (*CipAttributeEncodeInMessage)(const void *const data,
                                             ENIPMessage *const outgoing_message);
 
+/** @brief self-describing data decoding for CIP types */
+typedef int (*CipAttributeDecodeInMessage)(const void *const data,
```

Comment:
You are not decoding into a message but from a message, therefore this should be called CipAttributeDecodeFromMessage